src/debug/dwarf/info.cr

```diff
@@ -0,0 +1,141 @@
+require "../dwarf"
+require "./abbrev"
+
+module Debug
+  module DWARF
+    struct Info
+      property unit_length : UInt32 | UInt64
+      property version : UInt16
+      property debug_abbrev_offset : UInt32 | UInt64
+      property address_size : UInt8
+      property! abbreviations : Array(Abbrev)
+
+      property dwarf64 : Bool
+      @offset : LibC::OffT
+      @ref_offset : LibC::OffT
+
+      def initialize(@io : IO::FileDescriptor, @offset)
+        @ref_offset = offset
+
+        @unit_length = @io.read_bytes(UInt32)
+        if @unit_length == 0xffffffff
+          @dwarf64 = true
+          @unit_length = @io.read_bytes(UInt64)
+        else
+          @dwarf64 = false
+        end
+
+        @offset = @io.tell
+        @version = @io.read_bytes(UInt16)
+        @debug_abbrev_offset = read_ulong
+        @address_size = @io.read_byte.not_nil!
+      end
+
+      alias Value = Bool | Int32 | Int64 | Slice(UInt8) | String | UInt16 | UInt32 | UInt64 | UInt8
+
+      def read_abbreviations(io)
+        @abbreviations = Abbrev.read(io, debug_abbrev_offset)
+      end
+
+      def each
+        end_offset = @offset + @unit_length
+        attributes = [] of {AT, FORM, Value}
+
+        while @io.tell < end_offset
+          code = DWARF.read_unsigned_leb128(@io)
+          attributes.clear
+
+          if abbrev = abbreviations[code - 1]? # abbreviations.find { |a| a.code == abbrev }
```

Comment:
The commented part is correct but slow, the actual is faster, but maybe wrong. This looks like working, but must be tested more.

A better note with explanation should be added.